### ISSUE TITLE
ui-sref does not work on md-tab when keyboard navigation is used

### ISSUE THREAD

**Issue by xsegrity** [angular/material#5819]
Navigating via the keyboard (e.g. using arrows and then selecting tab with the spacebar) does not sync the state properly when using ui-sref with md-tab.

```html
<md-tabs>
  <md-tab label="Tab A" ui-sref="taba" md-active="$state.is('taba')"></md-tab>
  <md-tab label="Tab B" ui-sref="tabb" md-active="$state.is('tabb')"></md-tab>
  <md-tab label="Tab C" ui-sref="tabc" md-active="$state.is('tabc')"></md-tab>
</md-tabs>
```

Changing this line of code in the keydown function seems to fix the problem but I am not sure if it causes other issues.
(https://github.com/angular/material/blob/master/src/components/tabs/js/tabsController.js#L277)

From:
```javascript
if (!locked) ctrl.selectedIndex = ctrl.focusIndex;
```

To:
```javascript
if (!locked) select(ctrl.focusIndex);
```

**Comment by Splaktar** [comment]
https://github.com/angular/material/issues/2344 was reopened to address this.

**Comment by ThomasBurleson** [comment]
Closing as duplicate.
